lib/ansible/modules/database/postgresql/postgresql_slot.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: postgresql_slot
+short_description: Add or remove slots from a PostgreSQL database.
+description:
+  - Adds or removes physical or logical slots from a PostgreSQL database.
+version_added: "2.8"
+options:
+  slot_name:
+    description:
+      - name of the slot to add or remove
+    required: true
+  type:
+    description:
+      - slots come in two distinct flavors, physical and logical
+    required: false
+    default: physical
+    choices: [ "physical", "logical" ]
+  db:
+    description:
+      - Name of the database where you're connecting in order to add or remove only the logical slot to/from
+    required: false
+  decoder:
```

Comment:
Would it be better to stick with the naming convention used in Postgres [docs](https://www.postgresql.org/docs/current/view-pg-replication-slots.html) and use `plugin` or `output_plugin` instead of `decoder`